test/test_nested_tensor_masking.py

```diff
@@ -39,312 +45,995 @@ def test_gen_nested_tensor(self):
 
     #
     # Group of tests to test to_tensor_mask() 
-    # 
-    def test_to_tensor_mask_nt_dim_2(self):
-        nt1 = nt.nested_tensor([
-            torch.tensor([1, 2]),
-            torch.tensor([3, 4])
+    #
+    def test_empty_nt(self):
+        a = nt.nested_tensor([])
+        tensor, mask = a.to_tensor_mask()
+
+        self.assertEqual(mask, torch.tensor([]))
+        self.assertEqual(tensor, torch.tensor([]))
+        check_tensor_options(self, tensor, a)
+
+        a = nt.nested_tensor([
+            nt.nested_tensor([])
         ])
-        exp_tensor = torch.tensor([[1, 2], [3, 4]])
-        exp_mask = torch.tensor(True)
 
-        res_tensor, res_mask = nt1.to_tensor_mask()
-        self.assertEqual(res_mask, exp_mask)
-        self.assertEqual(res_tensor, exp_tensor)
+        tensor, mask = a.to_tensor_mask()
+        self.assertEqual(mask, torch.tensor([[]]))
+        self.assertEqual(tensor, torch.tensor([[]]))
+        check_tensor_options(self, tensor, a)
 
-    def test_to_tensor_mask_nt_dim_4(self):
-        original_nt2 = nt.nested_tensor([
-            nt.nested_tensor([
+        a = nt.nested_tensor([
+            nt.nested_tensor([]),
+            nt.nested_tensor([])
+        ])
+
+        tensor, mask = a.to_tensor_mask()
+        self.assertEqual(mask, torch.tensor([[], []]))
+        self.assertEqual(tensor, torch.tensor([[], []]))
+        check_tensor_options(self, tensor, a)
+
+    #TODO once .to_list() bug fixed
+    def test_empty_tensor(self):
+        #a = nt.nested_tensor([
+        #        torch.tensor([])
+        #    ])
+        #self.assertRaisesRegex(RuntimeError, "Empty tensors are not yet supported.", lambda: a.to_tensor_mask())
+
+        a = nt.nested_tensor([
                 nt.nested_tensor([
-                    torch.tensor([1, 2])
-                ]),
-            ]),
-            nt.nested_tensor([
+                    torch.tensor([])
+                ])
+            ])
+        self.assertRaisesRegex(RuntimeError, "Empty tensors are not yet supported.", lambda: a.to_tensor_mask())
+
+    def test_single_scalar(self):
+        a = nt.nested_tensor([
+                torch.tensor(1, dtype=torch.uint8)
+            ])
+        tensor, mask = a.to_tensor_mask()
+        self.assertEqual(tensor, torch.tensor([1], dtype=torch.uint8))
+        self.assertEqual(mask, torch.tensor(True))
+        check_tensor_options(self, tensor, a)
+
+        tensor, mask = a.to_tensor_mask(mask_dim=0)
+        self.assertEqual(tensor, torch.tensor([1], dtype=torch.uint8))
+        self.assertEqual(mask, torch.tensor(True))
+        check_tensor_options(self, tensor, a)
+
+        tensor, mask = a.to_tensor_mask(mask_dim=1)
+        self.assertEqual(tensor, torch.tensor([1], dtype=torch.uint8))
+        self.assertEqual(mask, torch.tensor(True))
```

Comment:
The mask dimension here is 0 right? Why not 1?

Reply:
I'm somewhat assuming this is a typo. A helper function that parameterizes on mask_dim could have prevented that.

Reply:
yeah, its a bug

Reply:
fixed